e2e/benchmarks/services_test.go

```diff
@@ -0,0 +1,276 @@
+	for _, s := range generateServices(N, true) {
+		testutils.RegisterConsulService(t, srv, s, testutil.HealthPassing)
+	}
+	return makeConfig(t, srv.HTTPAddr, false)
+}
+
+// add generated services as separte services
+func setupMultiServices(t testing.TB, srv *testutil.TestServer, N int) (string, func() error) {
+	for _, s := range generateServices(N, false) {
+		testutils.RegisterConsulService(t, srv, s, testutil.HealthPassing)
+	}
+	return makeConfig(t, srv.HTTPAddr, false)
+}
+
+// generate service instance entries
+func generateServices(n int, instances bool) []testutil.TestService {
+	baseport := 30000
+	services := make([]testutil.TestService, n)
+	for i := 0; i < n; i++ {
+		name, id := "service", fmt.Sprintf("svc_%d", i)
+		if instances {
+			name, id = fmt.Sprintf("svc_%d", i), "instance"
+		}
+		services[i] = testutil.TestService{
+			Name:    name,
+			ID:      id,
+			Address: "127.0.0.2",
+			Port:    int(baseport + i),
+			Tags:    []string{},
+		}
+	}
+	return services
+}
+
+// writes the config files out to disk, returns path to them and cleanup func
+func makeConfig(t testing.TB, addr string, tls bool) (string, func() error) {
+	tempDir := fmt.Sprintf("%s%s", tempDirPrefix, "stress")
+	cleanup := testutils.MakeTempDir(t, tempDir)
+	modDir := filepath.Join(tempDir, "module")
+	testutils.MakeTempDir(t, modDir)
+
+	configPath := filepath.Join(tempDir, configFile)
+	err := writeConfig(configPath, configContents(addr, tls))
+	require.NoError(t, err)
+	modulePath := filepath.Join(modDir, "main.tf")
+	err = writeConfig(modulePath, nullModuleContents())
+	require.NoError(t, err)
+
+	return configPath, cleanup
+}
+
+// write the (config) contents to the path
+func writeConfig(configPath, contents string) error {
+	f, err := os.Create(configPath)
+	if err != nil {
+		return nil
+	}
+	defer f.Close()
+	config := []byte(contents)
+	_, err = f.Write(config)
+	return err
+}
+
+// not used but handy to keep around for when you need to check the added
+// services or instances in consul
+func showMeServices(t *testing.T, srv *testutil.TestServer) {
+	u := fmt.Sprintf("http://%s/v1/agent/services", srv.HTTPAddr)
+	resp := testutils.RequestHTTP(t, http.MethodGet, u, "")
+	b, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		panic(err)
+	}
+	fmt.Println(string(b))
+	defer resp.Body.Close()
+}
+
+// returns templated contents of config file
+func configContents(address string, tls bool) string {
+	pwd, err := os.Getwd()
+	if err != nil {
+		panic(err)
+	}
+	return fmt.Sprintf(`
+log_level = "WARN"
+
+consul {
+  address = "%s"
+  tls {
+	  enabled = %t
+	  verify = false
+  }
+}
+
+driver "terraform" {
+	path = "%s"
+	backend "local" {}
+}
+
+terraform_provider "local" {}
+
+task {
+  name = "stress-test"
+  source = "../../tmp_stress/module"
+  providers = ["local"]
+  services = ["stress"]
+}
+
+`, address, tls, pwd)
+}
+
+// null resource, services module
+func nullModuleContents() string {
```

Comment:
Suggestion only if you like, I think it could be worth adding the actual null module files. I could see myself using this in the future.

Reply:
You mean including them in a directory like the e2e/test_modules/ ones? Instead of it being generated?

Reply:
Yes, sorry for the lack of clarity! Yes, including the files in 'e2e/test_modules/' instead of generating them.

Reply:
I'll do this but I'll do it later. Spent all my time getting the benchmarks to work right and forgot this and don't have time today and we want to go ahead and merge this. I'll add a note to refactor this our next week as it won't take long.